### PR TITLE
feat: add agents page with list of agents

### LINKED ISSUE #1418: Install agent page
- After installing an agent, popup dialog with copy and paste instructions for installing an agent with the tokens embedded into the scripts

### PR DESCRIPTION
**Pull request by mainawycliffe**
Closes #1418

### CODE REVIEW

File: src/components/Agents/List/AgentsTableColumns.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    accessorKey: "username",
+    minSize: 150,
+    enableSorting: true
+  },
+  {
+    header: "TLS",
+    accessorKey: "tls",
+    minSize: 80,
+    enableSorting: true
+  },
```
Comment (moshloop) [inline src/components/Agents/List/AgentsTableColumns.tsx:53]: We only need name, created/updated_at, created_by, labels, and the counts as per : https://github.com/flanksource/duty/blob/main/views/022_agents.sql#L1

**Comment by mainawycliffe** [comment]
## TODO

- [ ] Ensure that the latest changes on Duty are deployed to test environment for this to work (agent_summary view is missing)
- [ ] Get the helm install command and fine-tune installation instructions from Yash, once ready

File: src/App.tsx [hunk 2]
```diff
@@ -117,6 +119,13 @@ const navigation: NavigationItems = [
     icon: FaTasks,
     featureName: features.playbooks,
     resourceName: tables.database
+  },
+  {
+    name: "Agents",
```
Comment (moshloop) [inline src/App.tsx:124]: These should go into settings

File: src/components/Agents/Add/AddAgentForm.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+  onSuccess: (agent: GeneratedAgent) => void;
+};
+
+export default function AgentForm({ isOpen, onClose, onSuccess }: Props) {
+  const { mutate: upsertAgent, isLoading } = useUpsertAgentMutations({
+    onSuccess: (data) => {
+      toastSuccess("Agent saved");
```
Comment (moshloop) [inline src/components/Agents/Add/AddAgentForm.tsx:21]:
```suggestion
      toastSuccess("Agent created");
```

File: src/components/Agents/Add/AddAgentForm.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      onSuccess(data);
+    },
+    onError: (error) => {
+      toastError(error.message);
+    }
+  });
+
+  return (
+    <Modal
+      title={"Add Agent"}
```
Comment (moshloop) [inline src/components/Agents/Add/AddAgentForm.tsx:31]:
```suggestion
      title={"Create new agent"}
```

File: src/components/Agents/Add/AddAgentForm.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                </div>
+              </div>
+            </div>
+            <div className="flex items-center justify-between py-4 px-5 bg-gray-100">
+              <Button
+                icon={
+                  isLoading ? <FaSpinner className="animate-spin" /> : undefined
+                }
+                type="submit"
+                text={isLoading ? "Saving ..." : "Save"}
```
Comment (moshloop) [inline src/components/Agents/Add/AddAgentForm.tsx:67]:
```suggestion
                text={isLoading ? "Saving ..." : "Next"}
```

File: src/components/Agents/Add/AddAgentForm.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+                  <FormikTextInput name="name" label="Name" required />
+                  <FormikKeyValueMapField
+                    name="properties"
+                    label="Properties"
+                  />
+                </div>
+              </div>
+            </div>
+            <div className="flex items-center justify-between py-4 px-5 bg-gray-100">
+              <Button
```
Comment (moshloop) [inline src/components/Agents/Add/AddAgentForm.tsx:62]: Can we align the button right so it looks more like a Wizard

File: src/components/Agents/InstallAgentModal.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+export default function InstallAgentModal({
+  isOpen,
+  onClose,
+  generatedAgent
+}: Props) {
+  return (
+    <Modal
+      title={"Instructions to install the agent"}
```
Comment (moshloop) [inline src/components/Agents/InstallAgentModal.tsx:19]:
```suggestion
      title={"Installation Instructions"}
```

File: src/components/Agents/InstallAgentModal.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      bodyClass="flex flex-col w-full flex-1 h-full overflow-y-auto"
+    >
+      <div className="flex flex-col flex-1 p-4 gap-4 overflow-y-auto">
+        <p>1. Add the Flanksource Helm repository to your local Helm client</p>
+        <CodeBlock code="helm repo add flanksource https://flanksource.github.io/charts" />
+        <p>2. And run repo update </p>
+        <CodeBlock code={"helm repo update"} />
+        <p>3. And finally install the agent with the following command.</p>
+        <CodeBlock
+          code={`helm install mc-agent flanksource/mission-control-agent -n "<namespace>"  --create-namespace --set upstream.createSecret=true --set upstream.host=$host --set upstream.username=${generatedAgent.username} --set upstream.password=${generatedAgent.access_token}`}
```
Comment (moshloop) [inline src/components/Agents/InstallAgentModal.tsx:31]: The host needs to come from the org metadata (or the current location)

Namespace should also be "mission-control-agent" not `<namesspace>`

File: src/components/Agents/List/AgentsTableColumns.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    header: "Created At",
+    minSize: 150,
+    enableSorting: true,
+    cell: DateCell
+  },
+  {
+    header: "Updated At",
+    minSize: 150,
+    enableSorting: true,
+    cell: DateCell
```
Comment (moshloop) [inline src/components/Agents/List/AgentsTableColumns.tsx:45]: I don't think these should have min Sizes

File: src/components/Forms/Formik/FormikKeyValueMapField.tsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+            />
+            <Button
+              className="btn-white"
+              onClick={() => handleRemove(index)}
+              icon={<FaTrash />}
+            />
+          </div>
+        ))}
+      </div>
+      <div className="flex flex-row gap-2 justify-end">
```
Comment (moshloop) [inline src/components/Forms/Formik/FormikKeyValueMapField.tsx:94]: I think this should be left aligned, and no outline buttons (same for the remove buttons)

**Pull request merged**